pkg/virtctl/objectgraph/objectgraph_test.go

```diff
@@ -0,0 +1,219 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors.
+ *
+ */
+
+package objectgraph_test
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"go.uber.org/mock/gomock"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	k8stesting "k8s.io/client-go/testing"
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+	kubevirtfake "kubevirt.io/client-go/kubevirt/fake"
+	kvtesting "kubevirt.io/client-go/testing"
+
+	"kubevirt.io/kubevirt/pkg/pointer"
+	"kubevirt.io/kubevirt/pkg/virtctl/testing"
+)
+
+var _ = Describe("Object graph command", func() {
+	const (
+		objectGraphCommand = "objectgraph"
+		vmName             = "testvm"
+	)
+
+	var (
+		virtClient *kubevirtfake.Clientset
+	)
+
+	BeforeEach(func() {
+		ctrl := gomock.NewController(GinkgoT())
+		virtClient = kubevirtfake.NewSimpleClientset()
+		kubecli.MockKubevirtClientInstance = kubecli.NewMockKubevirtClient(ctrl)
+		kubecli.GetKubevirtClientFromClientConfig = kubecli.GetMockKubevirtClientFromClientConfig
+		kubecli.MockKubevirtClientInstance.EXPECT().VirtualMachine(metav1.NamespaceDefault).Return(virtClient.KubevirtV1().VirtualMachines(metav1.NamespaceDefault)).AnyTimes()
+		kubecli.MockKubevirtClientInstance.EXPECT().VirtualMachineInstance(metav1.NamespaceDefault).Return(virtClient.KubevirtV1().VirtualMachineInstances(metav1.NamespaceDefault)).AnyTimes()
+	})
+
+	expectVMObjectGraphGet := func(opts *v1.ObjectGraphOptions, rootNode v1.ObjectGraphNode, errors ...error) {
+		virtClient.PrependReactor("get", "virtualmachines/objectgraph", func(action k8stesting.Action) (handled bool, ret runtime.Object, err error) {
```

Comment:
@fossedihelm Do you consider this better/less complex compared to simply using the mock client?

Reply:
I did not mean to use the prepend reactor :)
BTW if you prefer the mock for the moment, I will not block it,
but the mock is more prone to false positives

Reply:
IIRC then fake client with subresource APIs is always problematic?

Reply:
IDK, need to check :)